### PR TITLE
Improve signature code generation

### PR DESCRIPTION
**Pull request by danielfriis**

### CODE REVIEW

File: app/controllers/signature_email_deliveries_controller.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class SignatureEmailDeliveriesController < ApplicationController
+  load_resource :account
+  load_and_authorize_resource :signature
+
+  def create
+    # TODO: Handle failures and add flash
+    SignatureMailer.delivery(@signature).deliver_now
```
Comment (kaspernj) [inline app/controllers/signature_email_deliveries_controller.rb]: I would argue to always use `.deliver_later`. Just set ActiveJob to use the `inline`-mode which will execute it right away regardless.

Then when you actually want to change to Sidekiq or DelayedJob later, you don't have to change anything in the code :-)

File: app/mailers/signature_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class SignatureMailer < ApplicationMailer
+  def delivery(signature)
+    @signature = signature
+    mail to: @signature.email, subject: "Signature test"
```
Comment (kaspernj) [inline app/mailers/signature_mailer.rb]: Translate `Signature test`?

File: app/views/layouts/mailer.html.haml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+%hmtl
+  %body
+    = yield
```
Comment (kaspernj) [inline app/views/layouts/mailer.html.haml]: Missing newline

File: app/views/layouts/mailer.text.haml [hunk 4]
```diff
@@ -0,0 +1 @@
+= yield
```
Comment (kaspernj) [inline app/views/layouts/mailer.text.haml]: Here as well

File: config/initializers/mail_setup.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# Change mail delvery to either :smtp, :sendmail, :file, :test
+ActionMailer::Base.delivery_method = :smtp unless Rails.env.test?
+ActionMailer::Base.smtp_settings = {
```
Comment (kaspernj) [inline config/initializers/mail_setup.rb]: Couldn't you just buy Mailgun over Heroku and avoid all this? :-)

Comment (danielfriis) [inline config/initializers/mail_setup.rb]: Not if I want to test in development? :)

File: app/views/layouts/mailer.text.haml (hunk 4 above)
Comment (danielfriis) [inline app/views/layouts/mailer.text.haml]: Fixed 2858312

File: app/views/layouts/mailer.html.haml (hunk 3 above)
Comment (danielfriis) [inline app/views/layouts/mailer.html.haml]: Fixed 2858312

File: app/mailers/signature_mailer.rb (hunk 2 above)
Comment (danielfriis) [inline app/mailers/signature_mailer.rb]: Fixed 2858312

File: app/controllers/signature_email_deliveries_controller.rb (hunk 1 above)
Comment (danielfriis) [inline app/controllers/signature_email_deliveries_controller.rb]: Fixed 2858312

**Pull request merged**